lib/puppet/application/cert.rb

```diff
@@ -264,7 +264,10 @@ def main
       hosts = command_line.args.collect { |h| h.downcase }
     end
     begin
-      apply(@ca, :revoke, options.merge(:to => hosts)) if subcommand == :destroy
+      if subcommand == :destroy
+        signed_hosts = hosts - (@ca.waiting? & hosts)
```

Comment:
This can be shorted to just `hosts - @ca.waiting?`, as any host in `waiting?`, but not `hosts` will not be returned:

```ruby
irb(main):002:0> [1, 2] - [2, 3]
=> [1]
```

Reply:
Makes sense.  Corrected it.